Bug 1573192 - remove email validation on metadata.owner

MihaiTabara opened the pull request:
Bugzilla Bug: [1573192](https://bugzilla.mozilla.org/show_bug.cgi?id=1573192)

@djmitche Hm, I'm not so sure this is so straightforward. I see there are plenty of tests and usecases around the codebase that are, expectedly, assuming this to be an email:

```
services/notify/src/handler.js:**Owner:** ${task.metadata.owner}
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': '', // private email
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': '', // private email
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': 'test@test.com',
services/github/test/intree_test.js:      'tasks[0].task.metadata.owner': 'test@test.com',
services/hooks/test/api_test.js:      inputWithTriggerSchema.metadata.owner = 'test@test.org';
services/hooks/src/scheduler.js:      address: hook.metadata.owner,
ui/src/components/HookForm/index.jsx:        owner: hook.metadata.owner,
ui/src/components/HookForm/index.jsx:      hook.metadata.owner &&
ui/src/components/HookForm/index.jsx:              value={hook.metadata.owner}
ui/src/views/Tasks/ViewTask/index.jsx:                    <em>{task.metadata.owner}</em>
```

While some of them may be ignored, there are some other occurrences which worry me. For example, within https://github.com/taskcluster/taskcluster/blob/master/services/hooks/src/scheduler.js#L151 the `createEmail` is used above in `sendFailureEmail`. If the `metadata.owner` is no longer an email, taskcluster will experience an issue in sending that but the error will likely raise a 510/511 visible only on server-side. Are we okay with this?

What do you think?

escapewindow commented:
We're seeing this on the xpi automation repos (failed taskcluster checks because bors doesn't have a valid email address). Any chance we can get some movement here?

MihaiTabara commented:
@escapewindow sorry, this slipped my radar once again. Added to my todo list for this week. Thanks for resurfacing it.

MihaiTabara commented:
@djmitche sorry for huge delays in updating this. mind if you re-review this? Thanks.

djmitche commented on changelog/bug-1573192.md:
```diff
@@ -0,0 +1,4 @@
+level: minor
+reference: bug 1573192
+---
+Remove email validation on metadata.owner
```
```suggestion
A task's `metadata.owner` is no longer required to have the form of an email address, as discussed in [RFC#153](https://github.com/taskcluster/taskcluster-rfcs/blob/master/rfcs/0153-remove-email-validation-for-metadata-owner.md).
```

djmitche commented on services/queue/schemas/v1/task-metadata.yml:
```diff
@@ -21,10 +21,10 @@ properties:
-      E-mail of person who caused this task, e.g. the person who did
-      `hg push`. The person we should contact to ask why this task is here.
+      Entity who caused this task, not necessarily a person with email who did
+      `hg push` as it could be automation bots as well. This entity we should
```
```suggestion
      `hg push` as it could be automation bots as well. The entity we should
```

MihaiTabara commented:
@djmitche thanks for addressing the nits and for merging this. I presume it's going to get deployed at the next Taskcluster deployment, right? Thank you!

escapewindow commented:
Thank you!
